### PR TITLE
Contributor docs: add `caniuse` usage guidance

### PR DESCRIPTION
**Pull request by ddbeck**
Fixes https://github.com/web-platform-dx/web-features/issues/1631

### CODE REVIEW

File: docs/guidelines.md [hunk 1]
```diff
@@ -223,3 +223,37 @@ Because `window` is both the global object and represents the browser window, th
 Use the most customary reference in each case.
 If you're not sure what's customary, look to high-profile published examples, such as those on MDN reference pages for the feature.
 ([#913](https://github.com/web-platform-dx/web-features/pull/913#discussion_r1572601975))
+
+## `caniuse` values
+
+The `caniuse` key references one or more [Can I Use](https://caniuse.com/) feature IDs.
+This is the part of a Can I Use URL after `https://caniuse.com/`.
+For example, the Can I Use feature ID for [Flexbox](https://caniuse.com/flexbox) is `flexbox`.
```
Comment (jamesnw) [inline docs/guidelines.md:231]: How about "A list of the `canisuse` keys also be found in the `caniuse` repo in the [`features-json` folder](https://github.com/Fyrd/caniuse/tree/main/features-json)."

File: docs/guidelines.md [hunk 2]
```diff
@@ -223,3 +223,37 @@ Because `window` is both the global object and represents the browser window, th
...
+- Do not set a `caniuse` value if the Can I Use feature is merely related to the feature.
+  For example, in [`grid.yml`](../features/grid.yml), do not set `caniuse: css-subgrid`.
+
+- Do not set a `caniuse` value if the top-level headline `status` would not be accurate with respect to the table on Can I Use.
+  For example, if Can I Use shows that a core browser set browser does not support a feature but web-features's status reports that the feature is Baseline high, then do not set the `caniuse` value for that feature.
+
+- Do not set a `caniuse` value if the top-level headline status's first-supported release (for example, the value of `status.support.safari`) differs from Can I Use's first-supported release by:
+
+  - More than one release for releases since 2020
+  - More than one year for releases before 2020
```
Comment (jamesnw) [inline docs/guidelines.md:247]: Is there a way we could programmatically test this?

**Review by jamesnw — APPROVED** [review]
Thanks! This is helpful!

File: docs/guidelines.md (hunk 1 above)
Comment (ddbeck) [reply docs/guidelines.md:231]: Good idea, But there are some weird exceptions, so I can do one better:

```suggestion
For example, the Can I Use feature ID for [Flexbox](https://caniuse.com/flexbox) is `flexbox`.
For a complete list of IDs, run `npx tsx ./scripts/caniuse.ts`.
```

File: docs/guidelines.md (hunk 2 above)
Comment (ddbeck) [reply docs/guidelines.md:247]: It's likely, yes. I've filed https://github.com/web-platform-dx/web-features/issues/1680.

**Review by captainbrosset — APPROVED** [review]

**Pull request merged**